pkgs/servers/sql/postgresql/generic.nix

```diff
@@ -54,10 +61,29 @@ let
       inherit hash;
     };
 
+    __structuredAttrs = true;
+
     hardeningEnable = lib.optionals (!stdenv'.cc.isClang) [ "pie" ];
 
-    outputs = [ "out" "lib" "doc" "man" ];
-    setOutputFlags = false; # $out retains configureFlags :-/
+    outputs = [ "out" "dev" "doc" "lib" "man" ];
+    outputChecks.out = {
+      disallowedReferences = [ "dev" "doc" "man" ];
+      disallowedRequisites = [
+        stdenv'.cc
+      ] ++ lib.optionals jitSupport [
+        llvmPackages.llvm.out
+        llvmPackages.llvm.dev
+      ];
+    };
+    outputChecks.lib = {
```

Comment:
Doesn't this have the downside of not being applied for all outputs?
AFAIU -dev is also not supposed to depend on stdenv'.cc, but this isn't checked anymore?